_posts/2019-06-06-gotham-0.4md

```diff
@@ -0,0 +1,167 @@
+
+gotham::tls::start(addr, config, || Ok(say_hello))
+```
+
+This simple tweak to the API makes it easy to handle TLS connections without
+having to learn an entire new set of APIs.
+
+## Middlewares
+
+Gotham 0.4 includes work on a few new middlewares designed to improve some
+common use cases, and fit in with common patterns when writing web services.
+Each of the middlewares below is available in either the core of Gotham 
+itself, or distributed as a separate crate on [crates.io](http://crates.io/).
+
+### Cookie Middleware
+
+Gotham 0.4 also includes a new middleware named `CookieParser`, which is
+designed to make dealing with cookies much simpler and in a much more 
+automatic fashion. All parsing of cookies is done via the de-facto
+[cookie](https://github.com/alexcrichton/cookie-rs) crate. 
+
+Unlike some of the other middlewares, the `CookieParser` middleware ships
+directly in Gotham's core library due to it being re-used in the existing
+`SessionMiddleware` workflows, and as it's such a common requirement when
+building a web service.
+
+To use this middleware, configure it as any other middleware inside your
+router:
+
+```rust
+// create a middleware pipeline from our middleware
+let pipeline = single_middleware(CookieParser);
+
+// construct a basic chain from our pipeline
+let (chain, pipelines) = single_pipeline(pipeline);
+
+// build a router with the chain & pipeline
+gotham::start("127.0.0.1:7878", build_router(chain, pipelines, |route| {
+    route.get("/").to(say_hello);
+}))
+```
+
+Then you will automatically have a `CookieJar` available on the `State`
+provided to your request, which you can easily access as needed:
+
+```rust
+CookieJar::borrow_from(&state)
+```
+
+If you have any existing middlewares dealing with cookies, maybe for
+things like authorization, you can now place them after the `CookieParser`
+in the middleware chain and they will then also have access to the new
+`CookieJar` entity. This allows you to remove any custom cookie parsing
+that you may have had to write in the past. One example of this is the
+existing `SessionMiddleware`, which will now look for any previously
+parsed cookies t avoid doing unnecessary work.
+
+An example of this middleware is available in the repository, so please
+do [check it out](https://github.com/gotham-rs/gotham/tree/master/examples/cookies/introduction).
+
+### Diesel Middleware
+
+[Diesel](https://github.com/diesel-rs/diesel) is a very popular ORM
+written in Rust, and there has been some work in progress for a compatible
+middleware for some time. Thanks to the work of `@colinbankier` this
+is now available via the `gotham_middleware_diesel` crate, which offers
+a convenient API for interacting with Diesel from Gotham.
+
+This middleware introduces a `Repo` struct, which is used as a layer
+between Diesel and Gotham to ensure that database interaction lives on
+a separate thread pool to the one connections are being served on. This
```

Comment:
I think technically it's not a "separate" thread pool - rather using a tokio feature that allows running blocking operations on the same thread pool without blocking the event loop (and thus other requests). I'm not sure how to word this succinctly, or whether this distinction is important here. 
Maybe just "database interaction does not block handling other requests" or similar?